istioctl/cmd/analyze.go

```diff
@@ -161,3 +176,48 @@ func serviceDiscovery() (bool, error) {
 		return false, fmt.Errorf("invalid argument value for discovery")
 	}
 }
+
+func colorPrefix(m diag.Message) string {
+	if !colorize {
+		return ""
+	}
+
+	prefix, ok := colorPrefixes[m.Type.Level()]
+	if !ok {
+		return ""
+	}
+
+	return prefix
+}
+
+func colorSuffix() string {
+	if !colorize {
+		return ""
+	}
+
+	return "\033[0m"
+}
```

Comment:
This looks great as-is. A couple of further thoughts for making this easily usable in other contexts:

* Should we put this somewhere more clearly accessible to other istioctl commands? 
* Should we combine these into a generic `func colorString(c Color, s string) string`?

Reply:
Let's get feedback for how this works in various OS and script vs interactive.

If we expect many commands will incorporate color we should bring in a color package as discussed in the work item, rather than trying to come up with this ourselves.  (See https://github.com/istio/istio/issues/17808 for discussion).

My intention was to do something trivial and get usability feedback that would let us know if we even want colors.  Unfortunately the convention seems to be to check if the output is a pipe and suppress color.  Everyone uses a library for that.  Adding a library has held this PR up for a long time.  I am thinking of taking out the automatic turning off of color for pipes so this will merge without top-level review.